Dependencies/xr/CMakeLists.txt

```diff
@@ -49,11 +47,12 @@ elseif (IOS)
     target_link_libraries(xr PRIVATE "-framework ARKit")
 else()
     if(WIN32)
-        target_include_directories(xr PRIVATE "Source/OpenXR/Windows")
+        target_include_directories(xr PRIVATE "Include/OpenXR/Windows")
     endif()
 
-    target_include_directories(xr PRIVATE "Dependencies/OpenXR-SDK/include")
-    target_include_directories(xr PRIVATE "Source/OpenXR")
+    target_include_directories(xr PUBLIC "Dependencies/OpenXR-MixedReality/openxr_preview/include")
+    target_include_directories(xr PUBLIC "Dependencies/OpenXR-MixedReality/shared/XrUtility")
```

Comment:
Publicly exposing an explicit OpenXR dependency is a pretty significant paradigm shift for XR, which to date has represented an opaque integration point allowing consuming logic to be completely decoupled from the underlying platform. An explicit OpenXR dependency isn't necessarily a bad thing, certainly in the long term; but this might be premature since it will open the door for comingled XR- and OpenXR-dependent code at a time when OpenXR isn't yet available on all the platforms we need to support.

I believe the intent here is to allow specific dependent consumers to access internal implementation details that aren't part of the broadly-supported core contract. If this is the case, it might be better to expose these as a separate secondary contract, similar to how we create [`Internal` contracts](https://github.com/BabylonJS/BabylonNative/blob/master/Plugins/NativeWindow/CMakeLists.txt#L20-L24) for certain Plugins that need to provide a deeper level of visibility to some, but not all, consumers. This would allow the core implementation-agnostic to remain intact and platform independent while still allowing extensions that need them to have more invasive access to the implementation details of the platform.

Reply:
Although an example extension isn't provided here, we want to write extensions that specifically take an openxr dependency. We also don't know that these openxr specific extensions will ever end up in the babylonnative repo. Some of the functionality in these extensions doesn't have webxr counterparts. There are also a wide variety of openxr types (XrInstance, XrSystemId, XrSession, frame time, etc) that need to be exposed to these OpenXR-specific extensions. We could refactor the xr component to somehow expose these component types, but I have the following concerns in doing this:
1. XR components won't map 1:1 across different platforms. Accessing a session on OpenXR may not have a parallel on ARCore/ARKit
1. Exposing generic components through the existing xr contract requires some sort of mechanism for obscuring types. This leaves more room for error around memory management/component lifecycles if we start handing around void pointers.
1. It will be difficult to maintain an xr component querying contract that is actually implemented by all platforms. We've seen a variety of instances where arcore/arkit work has already tacked in non-operable functionality in the OpenXR xr component. I would prefer to avoid doing this moving forward.

I agree with your observation that the intent is to access internal implementation details that aren't part of the broadly-supported core contract. Is there an example for where NativeWindowInternal is used today? It may be that Internal contracts would fulfill our needs but I can't say I fully understand the intention/implementation details of an internal contract based on that cmake definition.

Reply:
chatted offline, it made since to turn this into an internal contract. cmake files have been updated to reflect this ask.